apps/cpp_rpc/rpc_env.cc

```diff
@@ -20,141 +20,139 @@
  * \file rpc_env.cc
  * \brief Server environment of the RPC.
  */
+#include <cerrno>
 #include <tvm/runtime/registry.h>
-#include <errno.h>
-#ifndef _MSC_VER
-#include <sys/stat.h>
+#ifndef _WIN32
 #include <dirent.h>
+#include <sys/stat.h>
 #include <unistd.h>
 #else
 #include <Windows.h>
+#include <direct.h>
+namespace {
+  int mkdir(const char* path, int /* ignored */) { return _mkdir(path); }
+}
 #endif
+#include <cstring>
 #include <fstream>
-#include <vector>
 #include <iostream>
 #include <string>
-#include <cstring>
+#include <vector>
+#include <string>
 
-#include "rpc_env.h"
 #include "../../src/support/util.h"
 #include "../../src/runtime/file_util.h"
+#include "rpc_env.h"
+
+namespace {
+#if defined(__linux__) || defined(__ANDROID__)
+  const std::string untar_cmd = "tar -C ";
+#elif defined(_WIN32)
+  const std::string untar_cmd = "wsl tar -C ";
```

Comment:
I don't think it's ideal to have to force Windows users to enable WSL. Can we use Python [tarfile](https://docs.python.org/2/library/tarfile.html#examples) instead? Are there any other options?

Reply:
I considered some open-source C and C++ libraries, but shied away with license and time to implement reasons.

I think Python tarfile would be a good pick, especially if it comes packaged with python....as long as having python is not a burden.  I would like to test the speed though, mostly around python startup time and determine what is "acceptable" or not.

As a side note, it seems you can just send a precompiled dll (no tar), by changing the build_func:

`from tvm.contrib import cc`
....
`builder=autotvm.LocalBuilder(n_parallel=18, timeout=25, build_func=cc.create_shared)`

Reply:
I just did a few tests in powershell:
`Measure-Command {start-process "python tar-test.py" -Wait}`

Where tar-test.py is just: "import tarfile"

`Measure-Command {start-process "python wsl tar --help" -Wait}`

It doesnt look like python startup is much worse than "wsl tar".  Both around ~30ms.

How would you suggest this be implemented?  Generate a python script at runtime in the tmp folder?

Reply:
Looks like I do not need to create a python script at runtime.  It looks like something like this would work on windows:

python -m tarfile -e file.tar outdir/